fixes #142 (App.jsx, BgColorSidePanel.jsx, canvas.js): The line draw…

mondalsurojit opened the pull request:
fixes #142 (App.jsx, BgColorSidePanel.jsx, canvas.js): The line drawn gets un centered from the cursor when drawing further downwards on the canvas

## Type of change ☑️

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Code style update (formatting, local variables)

## Checklist: ☑️

- [X] My code follows the guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly wherever it was hard to understand.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] I have added things that prove my fix is effective or that my feature works.
- [X] Any dependent changes have been merged and published in downstream modules.


 
## How Has This Been Tested? ⚙️


https://github.com/singodiyashubham87/Draw-it-out/assets/114282267/ea7acfe1-dbb7-41a0-a4ba-fc3f91cf015a

mondalsurojit commented:
@singodiyashubham87 Sir I request you to review my PR...and I hope you will consider this as a level 3 issue...as this issue remained unsolved for a long time and it also took around 2 days just to figure out where the problem actually is...but finally I solved it...thank you in advance...

0xabdulkhalid commented on src/App.jsx:
```diff
@@ -162,10 +160,9 @@ function App() {
-              className={`whiteboard bg-slate-950 w-screen mt-[4vh] rounded-[0.6rem] shadow-md shadow-black dark:shadow-black dark:shadow-lg ${
-                isDrawing ? "cursor-crosshair" : "cursor-default pointer-events-none"
-              }
+              // id="draw"
+              className={`whiteboard bg-slate-950 rounded-[0.6rem] shadow-md shadow-black dark:shadow-black dark:shadow-lg ${isDrawing ? "cursor-crosshair" : "cursor-default pointer-events-none"
+                }
```
I would recommend you to remove `isDrawing` from `className`, because the `isDrawing` is always gonna be `true`. We don't need conditional rendering here.

So you can simply use `cursor-crosshair` for canvas `className`.

```suggestion
className="whiteboard bg-slate-950 rounded-[0.6rem] shadow-md shadow-black dark:shadow-black dark:shadow-lg cursor-crosshair"
```

0xabdulkhalid commented:
> [!NOTE]
> BTW, I mistakenly approved changes. I was about to request changes!

mondalsurojit commented:
@0xabdulkhalid I removed the extra conditional class...I hope now its okay...please see once...
@singodiyashubham87

0xabdulkhalid commented on src/App.jsx:
```diff
@@ -162,13 +160,9 @@ function App() {
-            `}
-              ref={canvasRef}
-            ></canvas>
+              // id="draw"
+              className={`whiteboard bg-slate-950 rounded-[0.6rem] shadow-md shadow-black dark:shadow-black dark:shadow-lg cursor-crosshair`}
```
You don't need to use backticks and curly brackets {``}.

React evaluates the expression inside the curly braces, even if it's just a static string, which involves a bit more processing.

It's not a best practice though, We only want to use backticks and curly braces when we need to include JavaScript expressions, such as conditional logic, variables, or template literals.

So, you need to use double quotes for static class names to avoid unnecessary processing.

Additionally you can remove the `id="draw"`, it's not needed.

You can copy paste the below suggestion,

```suggestion
className="whiteboard bg-slate-950 rounded-[0.6rem] shadow-md shadow-black dark:shadow-black dark:shadow-lg cursor-crosshair"
```

mondalsurojit replied:
I changed it now. Please check once, is it okay now...?

0xabdulkhalid reviewed (approved):
Good to go 🚀 

@singodiyashubham87, now this PR is ready to be merged.

singodiyashubham87 commented:
@mondalsurojit Resolve the conflicts.

mondalsurojit commented:
@singodiyashubham87 I have resolved the merge conflicts...now this branch is ready to be merged...

singodiyashubham87 commented:
@mondalsurojit Merge the main branch & push the changes.
# Your PR: 
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/c713f77f-5d0d-48aa-8358-02f5242a1b3c)
# Actual
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/1378ca23-a4bb-4b8a-a006-246099d74ff8)

singodiyashubham87 commented:
@mondalsurojit While resolving the conflicts you have removed the code that was added by other PRs to solve other issues, which is not the correct way to resolve the conflicts, you have to modify your code according to the main branch & not remove the main branch's code to make your's one working.

There are many functionalities broken.
1. the tour steps are missing, there is only one tour step which tells about the color palette & then directly tells to start drawing
2. After selecting shapes, when we again select the pencil tool, it defaults it to solid brush & when we try to change to dotted or dashed, it does not change.
3. Improper UI, the cross button is overlapping the canvas & there is very less space between the canvas & tools container
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/a3dc4911-83f4-40e1-b122-103637397006)

mondalsurojit commented:
@singodiyashubham87 Hello!
I had actually  checked everything before creating a PR, and everything was fine...Yes 2 bugs were noticed by me, which was caused due to some other previous PRs, and I believe those are directly or indirectly not connected to the code , I have written...

This is what my branch currently looks on my PC...
https://github.com/singodiyashubham87/Draw-it-out/assets/114282267/edd4072d-0e6d-4815-9a2c-28eb842f0c73

If you have saw my video, then you can see:
1) There are yet 3 steps, properly available one after the other in order. (p.s. still 1 step is missing, but I can't help, as that is caused due to some other problem)
2) (After selecting shapes, when we again select the pencil tool, it defaults it to solid brush & when we try to change to dotted or dashed, it does not change.) I have also demostrated this in the video.
3) The cursor on canvas is still a crosshair
4)The cross button is actually not overlapping...

I don't know, why you are facing such errors, on your side, it was not supposed to happen, as I have checked everything thoroughly before doing a PR (keeping aside 2 other bugs)....

Please check once, if everything is okay on your side

mondalsurojit commented:
@singodiyashubham87 What I feel is that, when I am making a branch from "main" and someone else is also making a branch form "main"...Both of us are simultaneously working on an issue, but if I create PR before him, and then he tries to create PR, the code will break or create merge conflicts, and this is unavoidable...somehow the whole scenario looks like a race condition...
For the proper merging, such that there is minimum to no merge conflicts, the whole workflow should be serialized...only one person working at a time on a PR, and they should have a deadline....

singodiyashubham87 commented:
@mondalsurojit Maybe the UI change is because of OS resolution, You are working on Windows & I'm working on Linux.
I would recommend you to copy the changes that you made & create a new clone of the project in your local machine then paste the changes & create a new fresh PR, which won't have any conflicts with the merged PRs.

singodiyashubham87 commented:
@mondalsurojit working on this?

mondalsurojit commented:
@singodiyashubham87 Please check once...If it doesn't work this time, I will create a new fork and try again...

singodiyashubham87 commented:
@mondalsurojit Conflicts, I recommend you to create a new fork.

mondalsurojit commented:
@singodiyashubham87 I have resolved it, actually very minute conflicts, like a portion I had deleted, or a function's name that I had changed....Please see once

singodiyashubham87 commented:
@mondalsurojit I've reviewed your PR & here are some changes that you need to make:

1. The cursor is a pointer in the latest changes but your still has the crosshair
2. The gap between canvas & tools is not appropriate
# Actual:
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/01cd2b69-7a69-4fba-8e74-499dd41901dc)
# Yours one: 
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/2d2c2c86-df39-469f-a1c9-7943bd2f3ac9)
3. In dark mode, the guidelines part looks different
# Actual:
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/d949b9d1-b62e-4736-8b22-6427cd913706)
# Yours: 
![image](https://github.com/singodiyashubham87/Draw-it-out/assets/93720428/167b4437-49cf-4c4a-99fc-725a475d4b7e)

singodiyashubham87 commented:
@mondalsurojit Well Done, Keep contributing :)
